### ISSUE TITLE
Relative pathname changed to absolute pathname when running updateComponent() - breaks model if the package is encapsulated

### ISSUE THREAD

**Issue by ceraolo** [OpenModelica/OpenModelica#9490]
### Description
I remember that years ago I opened a ticket similar to this one. Unfortunately, I cannot find it anymore; moreover, now I can supply precise details. That's why I create this new one.

I report two, closely related, issues:
1) OMEdit creates a wrong pathname when moving a component dragged from a subpackage
2) OMEdit displays the component with the wrong pathname as if it were correct

### Steps to Reproduce
Use the following mo file:
[RLC_PQtest.mo.txt](https://github.com/OpenModelica/OpenModelica/files/9733828/RLC_PQtest.mo.txt)

1) load the RLC_PQtest.mo
2) open RL_P
3) drag Submodels.Avg on RL_P: the code will contain the following correct row:
`Submodels.Avg avg;
`
4) move avg around; its code will be changed into:
`RLC_PQtest.Submodels.Avg avg;
`  This is presumably wrong (OMC considers it wrong), but OMEdit shows its icon, so the user is unaware of the mistake
5) try to simulate the model:. The following error message appears:
`[RLC_PQtest: 21:3-22:116]: [Class RLC_PQtest.Submodels.Avg not found in scope RL_P.]
`

Here I see two issues:
1)  OMEdit (only when avg is moved) prepends the package name to the model name, making it wrong
2) OMEdit shows the icon of a model that, when the model is run, cannot be found.

I  add that this error has happened on my second lesson at the University, so it is a very basic one, and confuses newbies a lot. I would recommend to solve it very soon.

### Expected Behavior
1) when we move a model in a diagram its pathname should never be changed
2) when a model has a wrong pathname (and therefore the container model cannot simulate) it should be represented by a red box,, not its icon.

### Version and OS
Win10
OM 1.19.2
MSL 4.0.0

 - OpenModelica Version: [e.g. `omc --version` or `Help->About OMEdit` from OMEdit]
 - OS: [e.g. Windows 10, 64 bit]
 - Versions of used Modelica libraries if applicable

**Comment by casella** [comment]
@ceraolo, as you know OMEdit relies on omc through an API to get information about Modelica models and to make changes to them. When you move around the component, this API function call is performed
```
updateComponent(avg,RLC_PQtest.Submodels.Avg,RLC_PQtest.RL_P,annotate=Placement(
visible=true, transformation=transformation(origin={18,-4}, extent={{-10,-10},{10,10}}, rotation=0))) 
```
and for some reason the internal representation of the model is changed from a relative path to an absolute path. This should probably be avoided, but is normally not an issue. The problem in your case is that your package `RLC_PQtest` is encapsulated, which actually prevents accessing the class Avg from the global scope.

I agree this shouldn't happen, but I do not understand why you make RLC_PQtest and encapsulated package. This is an advanced concept for information hiding that can be useful in certain contexts, but most certainly not for newbies.  Just remove the encapsulated keyword and everything's going to work fine.

We are completely rewriting the way OMEdit interacts with omc for these purposes, see #8425. This will be ready in about one month, within 1.21.0-dev. At that point, we will try to make sure that relative paths are not unnecessarily (and wrongly, in this case) changed into absolute ones when just editing the graphical layout of a model.

**Comment by ceraolo** [comment]
@casella, thank you for your comments.
Frankly, I don't remember why, (years ago), the encapsulated keyword was added. 

However, this has shown an issue, even though probably will not be observed frequently. In addition, it showed also issue 2 in my description.

So ok, I'll remove the "encapsulated" keyword, and that will solve my current problems.
We'll see what will happens when the new API is in place.

**Comment by ceraolo** [comment]
Now I remember the reason why I used encapsulated.
It is a good reason, that was conceived exactly for student use. But I can live without it for the time being.

I try to explain.
Consider a package P containing model A which uses as submodel model B. If, by chance, model B was created outside P, but is still in the workspace, model A runs well for the current session. 
If in the next session I open only P and try to run A, it does not, because B is not found. If I bring from school to home P, I believe to have brought with me all I need, but this is not true.

This is avoided if I define P encapsulated (and I import MSL in it): it must contain everything, except MSL models inside it. When I move my package to another computer I will be sure that I'll bring everything needed with me.

So, this is a use of the encapsulated keyword that I would not consider advanced. It is useful for newbies, instead. Nevertheless, as I said, I can live well without it, and I'll wait for the New API to come.

**Comment by adeas31** [comment]
Fixed in 3825633de08c1fb36d10d83ee6eb0c9087e287d0
